core/unit_test/TestTeam.hpp

```diff
@@ -1753,4 +1753,133 @@ struct TestRepeatedTeamReduce {
 
 }  // namespace Test
 
+namespace Test {
+
+struct SimpleTestValue {
+  int value;
+};
+
+struct SimpleFunctorReducer {
+  using value_type = SimpleTestValue;
+
+  KOKKOS_INLINE_FUNCTION
+  void init(value_type &init) const { init.value = 1; }
+
+  KOKKOS_INLINE_FUNCTION
+  void join(value_type &dst, value_type const &src) const {
+    dst.value *= src.value;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void final(value_type &) const {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, value_type &update) const {
+    update.value *= (i + 1);
+  }
+};
+
+struct SimpleReducer {
+  using reducer    = SimpleReducer;
+  using value_type = SimpleTestValue;
+
+  KOKKOS_INLINE_FUNCTION
+  SimpleReducer(value_type &val) : local(val) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void init(value_type &init) const { init.value = 1; }
+
+  KOKKOS_INLINE_FUNCTION
+  void join(value_type &dst, value_type const &src) const {
+    dst.value *= src.value;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void final(value_type &) const {}
```

Comment:
We should test with a non-trivial `final` as well.

Reply:
Modified the test to have something testable in `final`.